app/controllers/api/teams_controller.rb

```diff
@@ -63,18 +66,17 @@ def team
     @team ||= ::Team.find(params['id'])
   end
 
-  def find_teams(teams)
-    if query_param.present?
-      teams = finder(teams, query_param).apply
-    end
-    teams
+  def teams
+    @teams ||= current_user.teams
   end
 
-  def finder(teams, query)
-    Finders::TeamsFinder.new(teams, query)
+  def find_teams(params)
+    ::Teams::FilteredList.new(params: params, entries: teams).filter
```

Comment:
::Teams::FilteredList(current_user, params).new.fetch_entries

move everything to filtered list. in this controller, only fetch_entries calls FilteredList